src/pages/index.js

```diff
@@ -83,25 +89,122 @@ function handlePicturePopup(name, link) {
   popupImage.open(name, link);
 }
 
-const popupNewCard = new PopupWithForm(newCard, (data) => {
-  createCard(data);
-  popupNewCard.reset();
-  newCardFormValidator.disableSubmitButton();
-});
+const popupDelete = new PopupConfirmation(popupConfirmDelete, handleDeletePicture);
+
+function handleDeleteButton(cardId, element) {
+  popupDelete.open(cardId, element);
+}
+
+function handleDeletePicture(cardId, element) {
+  api
+    .deleteCard(cardId)
+    .then(() => {
+      element.remove();
```

Comment:
Needs correction: you should implement the function for removing cards in Card class and use it here. Don't remove the element directly here